Source/Libraries/FaultData/DataOperations/ConfigurationOperation.cs

```diff
@@ -1008,49 +999,47 @@ private void AddTriggerChannel(Channel powChannel, MeterDataSet meterDataSet)
                 TableOperations<Channel> channelTable = new TableOperations<Channel>(connection);
                 TableOperations<Series> seriesTable = new TableOperations<Series>(connection);
 
-                trendChannel.MeasurementType = measurementTypeTable.QueryRecordWhere("ID = {0}", trendChannel.MeasurementTypeID);
-                trendChannel.MeasurementCharacteristic = measurementCharacteristicTable.QueryRecordWhere("ID = {0}", trendChannel.MeasurementCharacteristicID);
+                string measCharacteristic = "";
+                string measurementType = "";
 
-                if (trendChannel.MeasurementCharacteristic.Name == "Trigger - RMS" || trendChannel.MeasurementCharacteristic.Name == "Trigger - Impulse")
+                if (Regex.IsMatch(trendChannel.Description, " RMS\\s*$", RegexOptions.IgnoreCase))
                 {
-                    trendChannel.MeasurementType = measurementTypeTable.QueryRecordWhere("Name = 'Voltage'");
-                    trendChannel.MeasurementTypeID = trendChannel.MeasurementType.ID;
+                    measCharacteristic = "RMS";
+                    measurementType = "Voltage";
                 }
-                else if (trendChannel.MeasurementCharacteristic.Name == "Trigger - I")
+                else if (Regex.IsMatch(trendChannel.Description, " Impulse\\s*$", RegexOptions.IgnoreCase))
                 {
-                    trendChannel.MeasurementType = measurementTypeTable.QueryRecordWhere("Name = 'Current'");
-                    trendChannel.MeasurementTypeID = trendChannel.MeasurementType.ID;
+                    measCharacteristic = "Instantaneous";
+                    measurementType = "Voltage";
                 }
-                else
+                else if (Regex.IsMatch(trendChannel.Description, " THD\\s*$", RegexOptions.IgnoreCase))
                 {
-                    trendChannel.MeasurementType = measurementTypeTable.QueryRecordWhere("Name = 'Analog'");
-                    trendChannel.MeasurementTypeID = trendChannel.MeasurementType.ID;
+                    measCharacteristic = "TotalTHD";
+                    measurementType = "Analog";
                 }
-
-                string measCharacteristic = "";
-
-                switch (trendChannel.MeasurementCharacteristic.Name)
+                else if (Regex.IsMatch(trendChannel.Description, " Unbalance\\s*$", RegexOptions.IgnoreCase))
+                {
+                    measCharacteristic = "Instantaneous";
+                    measurementType = "Analog";
+                }
+                else if (Regex.IsMatch(trendChannel.Description, " I.+\\s*$$", RegexOptions.IgnoreCase))
```

Comment:
On further inspection, I wonder if Tony was using `Trigger - I` for `3I0`. If so, this regex pattern wouldn't match it.